### PR TITLE
[VecOps] Use collection proxies for RVec I/O

### PR DESCRIPTION
**Pull request by eguiraud**
To be merged after v6.24 is branched.

- [x] add RVec to `TClassEdit::IsStdClass` (fixes `roottest_root_io_newstl_make`)

### CODE REVIEW

**Comment by eguiraud** [comment]
@phsft-bot build please

**Comment by eguiraud** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by eguiraud** [comment]
Test failure is unrelated (LLVM upgrade-related).

File: core/clingutils/src/TClingUtils.cxx [hunk 1]
```diff
@@ -4825,7 +4832,7 @@ ROOT::ESTLType ROOT::TMetaUtils::STLKind(const llvm::StringRef type)
 {
    static const char *stls[] =                  //container names
       {"any","vector","list", "deque","map","multimap","set","multiset","bitset",
-         "forward_list","unordered_set","unordered_multiset","unordered_map","unordered_multimap",0};
+         "forward_list","unordered_set","unordered_multiset","unordered_map","unordered_multimap", "RVec", 0};
```
Comment (pcanal) [inline core/clingutils/src/TClingUtils.cxx:4835]:
```suggestion
         "forward_list","unordered_set","unordered_multiset","unordered_map",
         "unordered_multimap", "RVec", 0};
```

File: io/io/src/TGenCollectionProxy.cxx [hunk 2]
```diff
@@ -1043,6 +1043,7 @@ void* TGenCollectionProxy::At(UInt_t idx)
    if ( fEnv && fEnv->fObject ) {
       switch (fSTL_type) {
       case ROOT::kSTLvector:
+      case ROOT::kROOTRVec:  // TODO will be unnecessary with RVec 2.0
```
Comment (pcanal) [inline io/io/src/TGenCollectionProxy.cxx:1046]: Can you expand the comment saying "why" it wont be necessary.

**Review by pcanal — APPROVED** [review]

File: core/clingutils/src/TClingUtils.cxx (hunk 1 above)
Comment (eguiraud) [reply core/clingutils/src/TClingUtils.cxx:4835]: I don't understand the suggestion: the line is not beyond ROOT's 120 columns, while clang-format would actually like for the list to be formatted this way:

```cpp
   static const char *stls[] = // container names
      {"any",           "vector",
       "list",          "deque",
       "map",           "multimap",
       "set",           "multiset",
       "bitset",        "forward_list",
       "unordered_set", "unordered_multiset",
       "unordered_map", "unordered_multimap",
       "RVec",          0};
```

File: io/io/src/TGenCollectionProxy.cxx (hunk 2 above)
Comment (eguiraud) [reply io/io/src/TGenCollectionProxy.cxx]: done

**Review by pcanal — APPROVED** [review]

**Pull request merged**